lib/uwb/UwbDevice.cxx

```diff
@@ -111,7 +111,15 @@ UwbDevice::OnSessionRangingData(UwbRangingData rangingData)
         return;
     }
 
-    // TODO: implement this
+    PLOG_VERBOSE << "Session with id " << rangingData.SessionId << " processing new ranging data";
+    std::vector<UwbPeer> peersData;
+    peersData.reserve(rangingData.RangingMeasurements.size());
+    for (const auto& peerData : rangingData.RangingMeasurements) {
+        UwbPeer data{ peerData };
+        PLOG_VERBOSE << "Peer data: " << data.ToString();
+        peersData.emplace_back(data);
```

Comment:
This should really use `push_back` instead, sine a copy is actually being made here. `emplace_back()` is intended for in-place pushing. That could be done here using `emplace_back(peerData)` and avoid creating the `UwbPeer` data object in the loop, but since you're logging it, what you've done makes sense.
```suggestion
        peersData.push_back(std::move(data));
```